lib/DXIL/DxilShaderModel.cpp

```diff
@@ -641,6 +641,7 @@ DXIL::NodeLaunchType ShaderModel::NodeLaunchTypeFromName(llvm::StringRef name) {
       .Case("broadcasting", DXIL::NodeLaunchType::Broadcasting)
       .Case("coalescing", DXIL::NodeLaunchType::Coalescing)
       .Case("thread", DXIL::NodeLaunchType::Thread)
+      .Case("mesh", DXIL::NodeLaunchType::Mesh)
```

Comment:
Should we have had a static_assert here as well to make sure this StringSwitch gets updated?

Reply:
do you just mean to throw in a static_assert that suggests the LaunchTypeNames hasn't been updated similar to this line? 

https://github.com/microsoft/DirectXShaderCompiler/blob/7f12aa652602bfc5f2510b04c6806fd10e52f3ee/lib/DXIL/DxilShaderModel.cpp#L633

That will only work if something is added to that list after mesh, which is likely, but I'm not sure it's what you're suggesting.

Reply:
I added what I think you want